components/ui/Button.tsx

```diff
@@ -70,21 +73,32 @@ export function CustomButton({
       style={[
         styles.buttonContainer,
         container,
-        disabled && styles.disabled,
+        loading ? styles.loading : disabled && styles.disabled,
         containerStyles,
       ]}
+      disabled={disabled}
       {...rest}>
       <View style={children && title ? [styles.childrenContainer] : {}}>
         {children}
       </View>
-      {title && (
+      {loading ? (
         <CustomText
           size={textSize}
           fontWeight="semibold"
           fontFamily="marfa"
           style={[text, styles.buttonText, textStyles]}>
-          {title}
+          Saving....
         </CustomText>
+      ) : (
+        title && (
+          <CustomText
+            size={textSize}
+            fontWeight="semibold"
+            fontFamily="marfa"
+            style={[text, styles.buttonText, textStyles]}>
+            {title}
+          </CustomText>
+        )
```

Comment:
instead of having this additional condition, can we have the button have different values for the title depending on whether it's loading or not, because these conditions look alike except for the text the user will see.

something like this,
```
  <CustomButton
        containerStyles={style.saveButton}
        variant="secondary"
        fullWidth
        title={loading?""Saving|"Save"}
        textSize="sm"
        onPress={handleSaveStatus}
        disabled={isLoading}
        loading={isLoading}/>

```